### PR TITLE
vim-patch:8.2.4720: ABB Rapid files are not recognized properly

### PR DESCRIPTION
**Pull request by clason**
vim-patch:8.2.4720: ABB Rapid files are not recognized properly

Problem:    ABB Rapid files are not recognized properly.
Solution:   Add checks for ABB Rapid files. (Patrick Meiser-Knosowski,
            closes #10104)
https://github.com/vim/vim/commit/b09c320039ad49e62d2e2d7f14ba47ee3ca0706a

### CODE REVIEW

**Comment by zeertzjq** [comment]
This is what `vim-patch.sh` lists:
```vim
  • v8.2.4720: Build file dependencies are incomplete
  • b09c320039ad49e62d2e2d7f14ba47ee3ca0706a: ABB Rapid files are not recognized properly
```
so I think `vim-patch:8.2.4720` is needed in the commit message to mark the patch as ported.

**Comment by zeertzjq** [comment]
The commit body also doesn't link the right commit. It should link <https://github.com/vim/vim/commit/0bbf09ca41382302493e5db51b01d2fbdc778586>.

**Comment by clason** [comment]
That's not how the commit history looks on Github, though...

**Comment by clason** [comment]
But I can manually merge the two commits in _our_ history.

**Comment by zeertzjq** [comment]
Tag v8.2.4720 points to <https://github.com/vim/vim/commit/0bbf09ca41382302493e5db51b01d2fbdc778586>

**Comment by clason** [comment]
Yes, but there is a _separate_ https://github.com/vim/vim/commit/b09c320039ad49e62d2e2d7f14ba47ee3ca0706a

I'll just pretend the previous commit is part of this.

File: runtime/autoload/dist/ft.vim [hunk 1]
```diff
@@ -112,6 +112,25 @@ func dist#ft#BindzoneCheck(default)
   endif
 endfunc
 
+" Returns true if file content looks like RAPID
+func IsRapid(sChkExt: string = ""): bool
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:116]:
```suggestion
func IsRapid(sChkExt = "")
```

File: runtime/autoload/dist/ft.vim [hunk 2]
```diff
@@ -415,6 +434,36 @@ func dist#ft#FTmm()
   setf nroff
 endfunc
 
+" Returns true if file content looks like LambdaProlog
+func IsLProlog(): bool
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:438]:
```suggestion
func IsLProlog()
```

File: runtime/autoload/dist/ft.vim [hunk 3]
```diff
@@ -415,6 +434,36 @@ func dist#ft#FTmm()
...
+  " skip apparent comments and blank lines, what looks like 
+  " LambdaProlog comment may be RAPID header
+  var l: number = nextnonblank(1)
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:441]:
```suggestion
  let l: number = nextnonblank(1)
```

File: runtime/autoload/dist/ft.vim [hunk 4]
```diff
@@ -415,6 +434,36 @@ func dist#ft#FTmm()
...
+  while l > 0 && l < line('$') && getline(l) =~ '^\s*%' # LambdaProlog comment
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:442]:
```suggestion
  while l > 0 && l < line('$') && getline(l) =~ '^\s*%' " LambdaProlog comment
```

File: runtime/autoload/dist/ft.vim [hunk 5]
```diff
@@ -415,6 +434,36 @@ func dist#ft#FTmm()
...
+    l = nextnonblank(l + 1)
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:443]:
```suggestion
    let l = nextnonblank(l + 1)
```

File: runtime/autoload/dist/ft.vim [hunk 6]
```diff
@@ -415,6 +434,36 @@ func dist#ft#FTmm()
   setf nroff
 endfunc
 
+" Returns true if file content looks like LambdaProlog
+func IsLProlog()
+  " skip apparent comments and blank lines, what looks like 
+  " LambdaProlog comment may be RAPID header
+  let l: number = nextnonblank(1)
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:441]:
```suggestion
  let l = nextnonblank(1)
```

File: runtime/autoload/dist/ft.vim [hunk 7]
```diff
@@ -112,6 +112,25 @@ func dist#ft#BindzoneCheck(default)
   endif
 endfunc
 
+" Returns true if file content looks like RAPID
+func IsRapid(sChkExt = "")
+  if sChkExt == "cfg"
```
Comment (zeertzjq) [inline runtime/autoload/dist/ft.vim:117]:
```suggestion
  if a:sChkExt == "cfg"
```

File: runtime/autoload/dist/ft.vim [hunk 8]
```diff
@@ -112,6 +112,25 @@ func dist#ft#BindzoneCheck(default)
   endif
 endfunc
 
+" Returns true if file content looks like RAPID
+func IsRapid(sChkExt = "")
```
Comment (clason) [inline runtime/autoload/dist/ft.vim:116]: Is that valid in legacy vimscript?

Comment (zeertzjq) [reply runtime/autoload/dist/ft.vim:116]: Yes. There is an example in `:h optional-function-argument`

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**